### PR TITLE
Checkstyle: Spaces instead of tabs

### PR DESCRIPTION
**Pull request by adamvoncorswant**
* Bump to checkstyle 6.2
* Add indentation checks
* Add import order checks

ping @nzroller or @softarn : I'm not sure how this repo is intended to be used. I'm sure about the checkstyle changes, but not about the pom-stuff.

### CODE REVIEW

**Comment by softarn** [comment]
It's this repo that should be used, the checkstyle exist in an other repo but it will be removed from there. Not sure about the pom stuff either.

What do you mean with Bump to checkstyle 6.2, don't see any related change?

I'm not sure import order check is a good idea. In 95% of the cases my IDE handles the imports, therefore for me it wouldn't improve the readability. If the problem is that some IDEs order them differently we should try to fix the IDE config before introducing the checkstyle rule, so it doesn't suddenly start bothering people without them knowing how to fix it.

File: basepom/pom.xml [hunk 1]
```diff
@@ -274,7 +274,7 @@
 						<dependency>
 							<groupId>com.izettle</groupId>
 							<artifactId>izettle-checkstyle</artifactId>
-							<version>1.0.11</version>
+							<version>1.0.11-SNAPSHOT</version>
```
Comment (freddd) [inline basepom/pom.xml]: Just a small comment, are we going to be using a SNAPSHOT or was this just for testing?

**Comment by adamvoncorswant** [comment]
There was a bug in checkstyle 5.x that assessed the wrong indentation level for expressions like:
```java
    object.method().member something something;
```
And only way to get passed that was by going above 6.
See https://github.com/iZettle/izettle-android/pull/3500

Regarding imports, that's exactly the import order we already have, and should not be a problem. I have already activated this exact configuration in the izettle-planet repo (not created a pulls request yet though, as I wanted this PR to be merged to I can reference a non-snapshot version of this code).

**Comment by nzroller** [comment]
@adamvoncorswant :+1: If we release v1.3 of the base pom then any project using it won't have to specify checkstyle configuration -- currently that's what the base pom is doing, just setting sensible defaults for plugins we commonly use. It's not a requirement for using the new checkstyle config though, that would be a release on the izettle-checkstyle jar.

@adamvoncorswant can you change the version on checkstyle/pom.xml to... `2.1-SNAPSHOT` ? There've already been a couple of releases since moving this code (v1.0.11 and v2.0.1)

Also note the checkstyle project does not have our base pom as parent because that would be a circular dependency (base pom depends on checkstyle which depends on base pom...). I'm not sure if you can release with the mvn release plugin (which has had a bug in subdir releasing on a git repo). May need to do it manually (fix version, commit it, build it, tag it, deploy it, etc.)

**Comment by adamvoncorswant** [comment]
Thanks @nzroller . Let's do this together tomorrow!

**Comment by adamvoncorswant** [comment]
So, @nzroller : I updated versions from `1.0.11-SNAPSHOT` to `2.1.11-SNAPSHOT`: is this how you meant it? If you're happy, please feel free to merge, and I'll try to perform a release of this to the artifactory!

**Comment by nzroller** [comment]
hehe, well I just wanted to avoid the library releases (2.0.2 was the latest) but heck why not `2.1.11` as the next release?

**Comment by nzroller** [comment]
I get this:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:2.12.1:checkstyle (validate) on project izettle-ejb-client: An error has occurred in Checkstyle report generation. Failed during checkstyle configuration: cannot initialize module TreeWalker - Property 'arrayInitIndent' in module Indentation does not exist, please check the documentation -> [Help 1]
```
So I'm guessing the checkstyle plugin version needs an upgrade?

**Comment by nzroller** [comment]
ok upgraded checkstyle to v6.2 and now get thousands on lines of the following:
```
/home/tim/projects/izettle/izettle-ejb-client/src/main/java/com/izettle/lib/kyc/RegistrationQuestion.java:21: 'method def modifier' have incorrect indentation level 8, expected level should be 4.
/home/tim/projects/izettle/izettle-ejb-client/src/main/java/com/izettle/lib/kyc/RegistrationQuestion.java:22: 'method def' child have incorrect indentation level 16, expected level should be 8.
/home/tim/projects/izettle/izettle-ejb-client/src/main/java/com/izettle/lib/kyc/RegistrationQuestion.java:23: 'method def rcurly' have incorrect indentation level 8, expected level should be 4.
Audit done.
```

Which is to be expected. Although the error messages is interesting since it sort of implies that if you adjust your tab width you could "pass" even with an indent rather than spaces?

**Comment by adamvoncorswant** [comment]
Ah, nice. I actually did change to these versions in each repo to get it working. Nice that it can be done here instead.

Yeah, one would assume that that would be possible, but after spending more time that I'd like on trying to get the indent-check to handle tabs properly, I had to give up. The reasoning is explained in
https://github.com/iZettle/izettle-android/pull/3500

So, long story short: As far as I understand, we have to use spaces if we want to use the checkstyle indent check.

I also see other benefits with using spaces:
* It's more or less a de facto standard, and a lot of tooling doesn't handle tabs as well as they should (looking at you vim, intellij, github diff view)
* The only ideological benefit of tabs is really that each coder can choose how he want to look at the code, as the indentation is then only a presentational setting. But if we want to set a maximum line length, this argument doesn't hold anymore since we force everyone to display a tab as 4 spaces.

**Comment by nzroller** [comment]
Merged!

**Pull request merged**